### DISCUSSION TITLE
Is Sandboxie-Plus going to nag users?

### DISCUSSION THREAD

**Discussion by bastik-1001** [sandboxie-plus/Sandboxie#2107] [General]
The latest language update contains the following string:

`Sandboxie <u>without</u> a valid supporter certificate will sometimes <b><font color='red'>pause for a few seconds</font></b>, to give you time to contemplate the option of <a href="https://sandboxie-plus.com/go.php?to=sbie-get-cert">supporting the project</a>.<br /><br />A <a href="https://sandboxie-plus.com/go.php?to=sbie-cert">supporter certificate</a> not just removes this reminder, but also enables <b>exclusive enhanced functionality</b> providing better security and compatibility.`

<img alt="Sandboxie nag screen" src="https://user-images.githubusercontent.com/12372772/232149486-3ddb222b-4c3e-4b99-8831-43c6bbf3596b.png" height="300px">

What made me wonder if users, will be getting nagged by Sandboxie?

What does it mean if it 'pauses for a few seconds'? How long are the durations of these pauses?

I remember someone not liking the rather subtle link in the main interface that asks to support the project, which is more than acceptable to me, but having Sandboxie getting in the way of its users is not something I find acceptable. Arguably, it is going to affect me, even though I could get a certificate, so I am not necessarily concerned for myself. If users shy away from the software, just because they get nagged can not be something that is good.

Don't get me wrong, development needs financial or other support.

**Comment by DavidXanatos** [comment]
A settled reminder once every couple of days should not be too annoying, see the code: https://github.com/sandboxie-plus/Sandboxie/blob/9df4fa911f5632f397e250fed8eca265b9b95c2a/SandboxiePlus/SandMan/Windows/SupportDialog.cpp#L64

```
        if (Days > 369) { // installed x days ago
            Interval = 3; // to show reminder in days
        } else if (Days > 120) {
            Interval = 5;
        } else if (Days > 80) {
            Interval = 7;
        } else {//if (Days > 40)	
            Interval = 10;
        }
```

I'm not sure if the selection of intervals is optimal, maybe it should be just once every 10 days after the first 40 days

**Reply by bastik-1001** [reply]
> not sure of the selection of intervals is optimal may be it should be just once every 10 days after the first 40 days

Those can easily be changed to whatever seems to fit, so that's something you need to evaluate.

What I still don't get, is what Sandboxie does when it 'pauses for a few seconds'. Just showing the reminder dialog? To me it sounded like it would behave like it is frozen.

**Reply by DavidXanatos** [reply]
No when being opened or restored from tray every x days it will show a window like this: https://youtu.be/N2_F-mAHOtI with a short countdown.

**Reply by bastik-1001** [reply]
That does not look as bad as it sounded when I read that it will pause.

**Reply by DavidXanatos** [reply]
if you add AlwaysShowReminder=y to the global section of your sandboxie.ini, in build 1.3.x, while not having a certificate set up, the reminder will be shown at every opportunity (neither 10 days nor after 40 days), useful for debugging :D

**Reply by bastik-1001** [reply]
Can indeed be quite useful to debug it.

**Reply by DavidXanatos** [reply]
i have thought a bit more about a good interval and came up with the following

            int Interval; // days
            if (Days > 730) Interval = 5;
            else if (Days > 365) Interval = 10;
            else if (Days > 180) Interval = 20;
            else Interval = 30;

**Reply by bastik-1001** [reply]
This seems to be reasonable, from once per month to once per week.

**Reply by Hiche3** [reply]
For those of us who use the classic version, and don't need the extra Plus features, is it really necessary for this frequent nag screen.

**Reply by DavidXanatos** [reply]
Well you know what to do to get rid of it ;) a lot of my work goes into the basic functionality, so its only fair to support that work as well, dont you think?

**Reply by ask2018** [reply]
> i have thought a bit more about a good interval and came up with the following
> 
> ```
>         int Interval; // days
>         if (Days > 730) Interval = 5;
>         else if (Days > 365) Interval = 10;
>         else if (Days > 180) Interval = 20;
>         else Interval = 30;
> ```

@DavidXanatos I don't want to start new issue for now, so I'll try to ask first here. Is this interval really still valid for Sandboxie Classic and does it work properly? Becuase this NAG actually works really odd for me. I've got installation date older than 870 days and for some sandboxed applications I can see the NAG basically every day, but for others I don't see it at all. I've also noticed some of the sandboxed application got issues to close itself properly when I click the close button.  The application windows closes itself, but the process is still running and I need to kill them in the sandboxie control or with taskmanager. But I'm not 100% sure if it is related to the NAG screen.

**Reply by DavidXanatos** [reply]
It should not be dependant on the started appliction. 
The close issue is not related to the nag screen.

**Comment by wedok** [comment]
Now in 2025 and Classic v5.7.12 i see nag screen every 24 hrs. :)

**Comment by NikoMuffin** [comment]
All software goes through levels of enshitification and now I see Sandboxie Plus is going this route.

It's really a shame.

**Comment by bastik-1001** [comment]
The issue #2929 is still open. At best, it would not be required and the reminder screen is just a button in the "about dialog" for users to learn why they might want to pay for it or how they can contribute. The next best thing is not having it show the reminder as often, but still offer a cheap and easy way to get rid of it.

**Reply by NikoMuffin** [reply]
Agreed. I would gladly pay $30 for Sandboxie (again, I paid for it many times before) for a lifetime license.

But the current lifetime costs are ludicrous
